Why is `user` optional coming out of `withPageAuthRequired()`?

### Checklist

- [X] The issue can be reproduced in the [sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [README](https://github.com/auth0/nextjs-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

I'm writing server-rendered pages with TypeScript, like this:

```tsx
import { withPageAuthRequired } from "@auth0/nextjs-auth0";
import { InferGetServerSidePropsType } from "next";

export const getServerSideProps = withPageAuthRequired({
  getServerSideProps: async ({ params }) => {
    return {
      props: {
        /* ... */
      },
    };
  },
});

type Props = InferGetServerSidePropsType<typeof getServerSideProps>;

const Page = ({ user }: Props) => {
  return <>Hello, {user.nickname}.</>; // <--- This `user` property comes from Auth0
};

export default Page;
```

Auth0's <a href="https://auth0.github.io/nextjs-auth0/modules/helpers_with_page_auth_required.html">withPageAuthRequired</a> will redirect the user if the session can't be resumed, or else it supplies a `user` prop and rendering proceeds. So, by the time execution gets to the rendering of the Page component, I expect that the `user` property is present and not null or undefined.

The code example above seems like it should work, but it causes a TypeScript error:

```
src/pages/example.page.tsx:17:20 - error TS18049: 'user' is possibly 'null' or 'undefined'.

17   return <>Hello, {user.nickname}.</>;
                      ~~~~
```

This comes from the `GetServerSidePropsResultWithSession` type:

https://github.com/auth0/nextjs-auth0/blob/d3d72afd94838a20a844603bc48d6f05f32317c8/src/helpers/with-page-auth-required.ts#L23

Is there a reason the `user` property is optional? Perhaps there's a case that my page components need to handle, that I hadn't considered?

Previous discussion: https://github.com/auth0/nextjs-auth0/issues/560

### Reproduction

1. Convert the sample app's [ssr.jsx](https://github.com/auth0-samples/auth0-nextjs-samples/blob/31315a0f1e166108c232761c8cab70ad55992f5e/Sample-01/pages/ssr.jsx) to TypeScript.
2. Enable TypeScript [strict mode](https://www.typescriptlang.org/tsconfig#strict).
3. Attempt to use a field on `user` (like `user.nickname`), see a TypeScript error:

```
./pages/ssr.tsx:32:14
Type error: 'user' is possibly 'null' or 'undefined'.

  30 |           <h6 className="muted">User prop</h6>
  31 |           <>{JSON.stringify(user, null, 2)}</>
> 32 |           <>{user.nickname}</>
     |              ^
  33 |         </div>
  34 |       </div>
  35 |     </>
```

### SDK version

2.1.0

### Next.js version

13.1.2

### Node.js version

18.12.1

--- a/src/helpers/with-page-auth-required.ts
+++ b/src/helpers/with-page-auth-required.ts
@@ -20,7 +20,7 @@ import { ParsedUrlQuery } from 'querystring';
  *
  * @category Server
  */
-export type GetServerSidePropsResultWithSession<P = any> = GetServerSidePropsResult<P & { user?: Claims | null }>;
+export type GetServerSidePropsResultWithSession<P = any> = GetServerSidePropsResult<P & { user: Claims }>;
 
 /**
  * A page route that has been augmented with {@link WithPageAuthRequired}.
@@ -51,8 +51,8 @@ export type PageRoute<P, Q extends ParsedUrlQuery = ParsedUrlQuery> = (
  *   async getServerSideProps(ctx) {
  *     // access the user session if needed
  *     // const session = await getSession(ctx.req, ctx.res);
- *     return { 
- *       props: { 
+ *     return {
+ *       props: {
  *         // customProp: 'bar',
  *       }
  *     };

